### PR TITLE
Feature/ode 123 sample kafka integration

### PR DESCRIPTION
**Pull request by tonychen091**
Documentation and instructions for running the sample application

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -68,3 +68,6 @@
 ### jpo-ode-plugins ###
 /jpo-ode-plugins/.apt_generated/
 /jpo-ode-plugins/target/
+jpo-ode-consumer-example/my-app/target/*
```
Comment (hmusavi) [inline .gitignore]: What's my-app for?

File: jpo-ode-consumer-example/pom.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+ <parent>
```
Comment (hmusavi) [inline jpo-ode-consumer-example/pom.xml]: This being a sample application should probably not be a child of jpo-ode. It should build on its own.

File: jpo-ode-consumer-example/pom.xml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  </parent>
+  <artifactId>consumer-example</artifactId>
+  <packaging>jar</packaging>
+  <version>1.0-SNAPSHOT</version>
+  <name>consumer-example</name>
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>3.8.1</version>
```
Comment (hmusavi) [inline jpo-ode-consumer-example/pom.xml]: Let's use junit 4

File: jpo-ode-consumer-example/src/main/java/com/consumerexample/app/ConsumerExample.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    	Option group_option = new Option("g", "group", true, "Consumer Group");
+    	group_option.setRequired(true);
+    	options.addOption(group_option);
+
+    	Option type_option = new Option("type", "type", true, "string|byte message type");
+    	type_option.setRequired(true);
+    	options.addOption(type_option);
+
+    	CommandLineParser parser = new DefaultParser();
+        HelpFormatter formatter = new HelpFormatter();
```
Comment (hmusavi) [inline jpo-ode-consumer-example/src/main/java/com/consumerexample/app/ConsumerExample.java]: Looks like indentation is a bit off in the nest 4 or so lines. Also, there is a mix of tabs and spaces in the files. Please check all the files to make sure only spaces are used.

File: jpo-ode-consumer-example/src/main/java/com/consumerexample/app/ConsumerExample.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+			KafkaConsumer<String, byte[]> byteArrayConsumer = new KafkaConsumer<String, byte[]>(props);
+
+			byteArrayConsumer.subscribe(Arrays.asList(topic));
+			System.out.println("Subscribed to topic " + topic);
+			while (true) {
+			 	ConsumerRecords<String, byte[]> records = byteArrayConsumer.poll(100);
+			    for (ConsumerRecord<String, byte[]> record : records) {
+			    	// Serialize the record value
+			    	SerializationUtils<J2735Bsm> serializer = new SerializationUtils<J2735Bsm>();
+			    	J2735Bsm bsm =  serializer.deserialize(record.value());
```
Comment (hmusavi) [inline jpo-ode-consumer-example/src/main/java/com/consumerexample/app/ConsumerExample.java]: Why does this throw ClassCastException in the ODE code? Did you figure that out?

File: jpo-ode-consumer-example/target/maven-archiver/pom.properties [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#Generated by Apache Maven
```
Comment (hmusavi) [inline jpo-ode-consumer-example/target/maven-archiver/pom.properties]: target directory and all files and sub-dirs should not be committed to Git.

File: jpo-ode-consumer-example/pom.xml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      <scope>test</scope>
+    </dependency>
+    <!-- https://mvnrepository.com/artifact/org.apache.kafka/kafka-clients -->
+    <dependency>
+      <groupId>org.apache.kafka</groupId>
+      <artifactId>kafka-clients</artifactId>
+      <version>0.10.1.0</version>
+    </dependency>
+    <!-- https://mvnrepository.com/artifact/org.springframework/spring-core -->
+    <dependency>
```
Comment (hmusavi) [inline jpo-ode-consumer-example/pom.xml]: Now that we have a standalone application, let's not depend on any jpo-ode-modules either. If you use java deserialization utils or org.apache.commons.lang3.SerializationUtils, you won't need jpo-ode-common.

Comment (tonychen091) [inline jpo-ode-consumer-example/pom.xml]: Any thoughts on what we should do about the J2735Bsm object?

File: jpo-ode-consumer-example/src/main/java/com/consumerexample/app/ConsumerExample.java (hunk 5 above)
Comment (tonychen091) [inline jpo-ode-consumer-example/src/main/java/com/consumerexample/app/ConsumerExample.java]: I'm not sure that I have an answer. I did try to do it in one line like a couple of examples, but I ran into an exception doing that.

File: jpo-ode-consumer-example/pom.xml (hunk 7 above)
Comment (tonychen091) [inline jpo-ode-consumer-example/pom.xml]: I also was not able to use any of the other serialization library to work. They kept giving me header errors

**Comment by tonychen091** [comment]
1. Merged with dev without any conflicts.
2. Agreed
3. Yep, missed some of the files there so I just removed the entire directory.

**Pull request merged**